UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs

```diff
@@ -849,6 +849,32 @@ public void ResetDamageAll()
 			{
 				bodyPart.ResetDamage();
 			}
+			
+		}
+
+		/// <summary>
+		/// Revives a dead player to full health.
+		/// </summary>
+		public void RevivePlayerToFullHealth(PlayerScript player)
+		{
+			Extinguish(); //Remove any fire on them.
+			ResetDamageAll(); //Bring their entire body parts that are on them in good shape.
+			healthStateController.SetOverallHealth(maxHealth); //Set the player's overall health to their race's maxHealth.
+			healthStateController.SetConsciousState(ConsciousState.CONSCIOUS); //They're allliiivveee!
```

Comment:
you have to restart the hearts manually, this doesn't work

Reply:
since it's determined in an update loop

Reply:
then why do defibs use the same logic if they don't work?
I took this straight out of the defib logic.

Reply:
I'm just saying it technically doesn't need it since the update loop will catch it, in CalculateOverallHealth